base_layer/core/src/base_node/comms_interface/inbound_handlers.rs

```diff
@@ -461,9 +488,19 @@ where B: BlockchainBackend + 'static
             source_peer
         );
 
-        let block = self.reconcile_block(source_peer.clone(), new_block).await?;
-        self.handle_block(block, Some(source_peer)).await?;
+        let block_result = self.reconcile_block(source_peer.clone(), new_block).await;
 
+        if block_result.is_err() {
+            let mut write_lock = self.list_of_reconciling_blocks.write().await;
+            write_lock.remove(&block_hash);
+        }
+        let block = block_result?;
+        let add_result = self.handle_block(block, Some(source_peer)).await;
+        {
+            let mut write_lock = self.list_of_reconciling_blocks.write().await;
+            write_lock.remove(&block_hash);
+        }
+        add_result?;
```

Comment:
Thanks, that's better - not critical but it would be cleaner to put reconcile and handle in a new function (something like `reconcile_and_add_block`) and handle the error case and locking in this outer function. Then you can use `?` for those calls

Reply:
I agree here, its cleaner to move to a single function